xprof/timeline.cpp

```diff
@@ -5,14 +5,23 @@
 #include <iomanip>  // set precision
 #include <iostream> // stdcout
 #include <stack>
-
+#include <string>
+#include <vector>
+#include <cstdint>
 #include "perfetto_prunned.pb.h"
 
 static perfetto_uuid_t gen_perfetto_uuid() {
   // Start at one, Look like UUID 0 is special
   static std::atomic<perfetto_uuid_t> uuid{1};
   return uuid++;
 }
+bool pushIfNotExists(std::vector<uint64_t>& vec, uint64_t value) {
+    if (std::find(vec.begin(), vec.end(), value) == vec.end()) {
+        vec.push_back(value);
+        return true;
+    }
+    return false;
+}
```

Comment:
```suggestion

static bool pushIfNotExists(std::vector<uint64_t>& vec, uint64_t value) {
  if (std::find(vec.begin(), vec.end(), value) == vec.end()) {
    vec.push_back(value);
    return true;
  }
  return false;
}
```
Ideally, it should use a set: https://cplusplus.com/reference/set/set/ I think